dataprofiler/labelers/data_processing.py

```diff
@@ -1151,6 +1156,11 @@ def process(self, data, labels=None, label_mapping=None, batch_size=32):
             raise ValueError('If `labels` are specified, `label_mapping` must '
                              'also be specified.')
 
+        if data.ndim > 1:
+            data = data.reshape(-1)
+        if labels is not None and labels.ndim > 1:
+            labels = labels.reshape(-1)
```

Comment:
Reshaping one but not the other seems like a recipe for a mismatch, also a very expensive operation `O(n)`.

Reply:
I do this since the input data needs to be a list, and the labels are a list of entities, which can be lists themselves. So if the data is given as a 2D array, I want to flatten it, but I don't want to always do this with the labels since they are supposed to be 2D in some sense.

Reply:
At least there should be a test for this or an error.